### DISCUSSION TITLE
i3 startup / zathura pdfviewer

### DISCUSSION THREAD

**Discussion by machielllo** [i3/i3#4884] [Q&A] [answered]
I'm running the following simple bash script:

#!/bin/bash
i3-msg "workspace 8Queueing; append_layout ~/.config/i3/queueingworkspace.json"

(zathura ~/queueing_book/queueing-book.pdf &)
(zathura -P 2 ~/queueing_book/course_overview.pdf &)
(alacritty -t "miniterm" --working-directory ~/queueing_book &)

Which works when run from within a terminal, but when I run it from within i3rc, either at start-up or with a keybinding, only alacritty is spawned, the other two screens become black and sometimes transparent.  The clock does disappear, so I feel like they get started, swallowed but then crash. When I xprop the black screen it just says i3-frame class.

I replicated the setup from within my i3 config, (with the use of i3-msg), without any succes. Same issue persists. I tried disabling picom, no succes.

Before I simply loaded the two PDF's on startup, into the right workspace, which worked without a problem. 

Any help is appreciated!

**Comment by Airblader** [comment]
I would guess that the issue is related to the `$PATH` in this case. But guess is all we can do.  Try redirecting stdout + stderr of those commands to a file to see if an error appears.

**Comment by machielllo** [comment]
The difference in the debug log is initially the order of acquiring d-bus and view ppi update, 
Then in the end I think it just stops rendering when I call it with a shortcut. it states:
debug: ../zathura/callbacks.c:235: cb_widget_screen_changed(): called
and then stops, while in the debug log when I call it from the command line it has a view lines about
render_job()

**Comment by machielllo** [comment, marked as answer]
Oh I found something. Disabling workspace_auto_back_and_forth fixed it.

**Reply by machielllo** [reply]
Semi fixes it. It only works if I try it in the workspace it self...
